### ISSUE TITLE
go/types: Document that the caller must populate desired fields of types.Info

### ISSUE THREAD

**Issue by creachadair** [golang/go#12787]
In the comment for the `go/types.(*Config).Check` method, it says:

```
Check type-checks a package and returns the resulting package object, the first error if any, and if info != nil, additional type information.
```

What it does not mention is that the caller must explicitly populate any of the map fields of this type that are to be filled in—it is not sufficient to simply pass a `*types.Info`. Ideally the documentation would make this explicit, say perhaps:

```
The type-checker populates each non-nil map field with the corresponding type information. If a given map field is nil, its information will be discarded.
```

or words to that effect. I'll be happy to send a PR for this (or similar), if we can agree on the wording.

**Comment by griesemer** [comment]
On it.

**Comment by creachadair** [comment]
It might be helpful to at least mention that in the Check documentation
anyway.

On Tue, Sep 29, 2015 at 12:49 PM, Robert Griesemer <notifications@github.com
> wrote:

> Closed #12787 <https://github.com/golang/go/issues/12787>.
>

**Comment by griesemer** [comment]
Will consider as part of another CL.
